### DISCUSSION TITLE
Error in using "TerminateControl" module

### DISCUSSION THREAD

**Discussion by avtarsinghh1991** [idaholab/moose#24626] [Q&A General] [answered]
Hello MOOSE experts,

I am trying to put control logic for terminating my simulation when postprocessor value reach certain value. However, I am getting following error.

```
*** ERROR ***
The parameter "_thm_problem" is being retrieved before being set.
```

Postprocessor

```
[Postprocessors]
  [average_surface_conc]
    type = SideAverageValue
    variable = c
    boundary = 'right'
  []
[]
```

```
[ControlLogic]
  [threshold]
    type = UnitTripControl
    condition = 'average_surface_conc < conc_ref'
    symbol_names = 'average_surface_conc'
    symbol_values = 'average_surface_conc'
  []

  [terminate]
    type = TerminateControl
    input = threshold:state
    termination_message = 'Surface concentration reached'
  []
[]
```

Any suggestions?

Best,
Avtar

**Comment by GiudGiud** [comment]
This control cannot be used outside of the thermal hydraulics module.

i made a change request this week to make this clear, it s very unfortunate timing that you tried to use it today, it will be clear in the documentation that that s not possible very soon

**Reply by avtarsinghh1991** [reply]
Thanks @GiudGiud 

Is there any other way to stop the simulation as soon as following postprocessor reaches certain value?

```
[Postprocessors]
  [average_surface_conc]
    type = SideAverageValue
    variable = c
    boundary = 'right'
  []
[]
```

**Reply by GiudGiud** [reply, marked as answer]
Use the Terminator It s designed for that purpose
https://mooseframework.inl.gov/source/userobjects/Terminator.html

**Reply by avtarsinghh1991** [reply]
Thanks @GiudGiud
